x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -162,16 +164,33 @@ class PackagePolicyService {
       }
       validatePackagePolicyOrThrow(packagePolicy, pkgInfo);
 
-      const registryPkgInfo = await Registry.fetchInfo(pkgInfo.name, pkgInfo.version);
+      const installablePackage: InstallablePackage = await Registry.fetchInfo(
+        pkgInfo.name,
+        pkgInfo.version
+      ).catch(async (error) => {
+        // Attempt to get an ArchivePackage object from the package info cache if the registry
+        // is not reachable. This will pull a bundled package if it was previously installed.
+        const archivePackage = await getArchivePackage({
```

Comment:
@kpollich any reason to not make this the default? the package should be installed at this point so we will have an archive package

Reply:
Good call. I swapped this around to default to an archive package, and call out to the registry if it doesn't exist.